### PR TITLE
[HLS-2615] Fixes to instrumentation indication

### PR DESCRIPTION
**Pull request by YochayCO**
What I mainly did here is make sure that only the first span is marked in the indication, like we do in python and nodejs

### CODE REVIEW

File: instrumentation/log4j/log4j-appender-2.17/library/src/main/java/io/opentelemetry/instrumentation/log4j/appender/v2_17/internal/LogEventMapper.java [hunk 1]
```diff
@@ -38,6 +42,21 @@
...
+  private static void markInstrumentationIndicator(AttributesBuilder attributes) {
+    Context parentContext = Context.current();
+    Span span = Span.fromContext(parentContext);
+    SpanContext parentSpanContext = span.getSpanContext();
+
+    if (!span.isRecording() || !parentSpanContext.isValid() || heliosInstrumentedIndicator) {
+      return;
+    }
+
+    attributes.put(HELIOS_INSTRUMENTED_INDICATION, "log4j");
```
Comment (levyoren) [inline instrumentation/log4j/log4j-appender-2.17/library/src/main/java/io/opentelemetry/instrumentation/log4j/appender/v2_17/internal/LogEventMapper.java]: don't we want to distinguish the two instrumentation? and set this to `log4j2`?

File: instrumentation/logback/logback-mdc-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/v1_0/LoggingEventWrapper.java [hunk 2]
```diff
@@ -19,6 +19,7 @@ final class LoggingEventWrapper implements ILoggingEvent {
   @Nullable private final LoggerContextVO vo;
 
   LoggingEventWrapper(ILoggingEvent event, Map<String, String> mdcPropertyMap) {
+    System.out.println("help");
```
Comment (levyoren) [inline instrumentation/logback/logback-mdc-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/v1_0/LoggingEventWrapper.java]: hope you're ok now :)
please remove.

**Review by levyoren — CHANGES_REQUESTED** [review]

File: instrumentation/logback/logback-mdc-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/v1_0/LoggingEventWrapper.java (hunk 2 above)
Comment (YochayCO) [reply instrumentation/logback/logback-mdc-1.0/library/src/main/java/io/opentelemetry/instrumentation/logback/v1_0/LoggingEventWrapper.java]: Yes it was rough 😆

**Review by levyoren — APPROVED** [review]

**Comment by YochayCO** [comment]
@levyoren if only the muzzle fails we can merge this. ok?

**Pull request merged**